### PR TITLE
support different time formats in Annotations.to_data_frame()

### LINKED ISSUE #12274: ENH: make use of time-stamps optional in `mne.Annotations.to_data_frame()`
### Describe the new feature or enhancement

Hi folks!

I find the `.to_data_frame()` method for annotations very practical but I realize that the default behavior of having onsets in timestamps is not always what is practical from a perspective of data analysis where you might want to keep selecting data based on times in seconds (float).

I was wondering how you would feel about including an enhancment that would make following snippet of code work:

```Python
annot = mne.Annotations(onset=[0, 10, 20], duration=[2] * 3, description='test')
annot_df = annot.to_data_frame(self, onset_as_time_stamp=False)
assert annot_df['onset'] == annot.onset
```

### Describe your proposed implementation

I would make conversion to time-stamps optional.

### Describe possible alternatives

I cannot think of another implementation. One can of course post-hoc edit the data frame but the point of a convenience method is to be convenient.

### Additional context

This can be helpful when analyzing continuous data in a fashion that does not involve epoching, e. g., in richly annotated long-term EEG recordings.

### PR DESCRIPTION
**Pull request by drammock**
~~it's working, but a bit hackish, so marking as draft while I clean it up.~~
closes #12274 

```pycon
In [1]: import mne
In [2]: foo = mne.Annotations(onset=[1,5,11], duration=[2, 2, 3], description=list('abc'))
In [3]: foo.to_data_frame(None)
Out[3]: 
   onset  duration description
0    1.0       2.0           a
1    5.0       2.0           b
2   11.0       3.0           c
In [4]: foo.to_data_frame("datetime")
Out[4]: 
                onset  duration description
0 1970-01-01 00:00:01       2.0           a
1 1970-01-01 00:00:05       2.0           b
2 1970-01-01 00:00:11       3.0           c
In [5]: foo.to_data_frame("ms")
Out[5]: 
   onset  duration description
0   1000       2.0           a
1   5000       2.0           b
2  11000       3.0           c
In [6]: foo.to_data_frame("timedelta")
Out[6]: 
            onset  duration description
0 0 days 00:00:01       2.0           a
1 0 days 00:00:05       2.0           b
2 0 days 00:00:11       3.0           c
```

### CODE REVIEW

**Comment by drammock** [comment]
@dengemann does this meet your needs?

**Comment by dengemann** [comment]
Looks awesome! This will facilitate interactive prototyping of analyses of signal windows with irregular duration. LGTM!

**Review by agramfort — APPROVED** [review]
❤️

**Comment by larsoner** [comment]
Nice, I think it just needs a `devel.rst` update, feel free to mark for merge-when-green once you push one @drammock !

File: mne/annotations.py [hunk 1]
```diff
@@ -442,9 +444,15 @@ def delete(self, idx):
         self.description = np.delete(self.description, idx)
         self.ch_names = np.delete(self.ch_names, idx)
 
-    def to_data_frame(self):
+    def to_data_frame(self, time_format="datetime"):
```
Comment (larsoner) [inline mne/annotations.py:447]:
```suggestion
    @fill_doc
    def to_data_frame(self, time_format="datetime"):
```

**Pull request merged**